src/profile/thetapr.jl

```diff
@@ -0,0 +1,103 @@
+
+"""
+    optsumj(os::OptSummary, j::Integer)
+
+Return an `OptSummary` with the `j`'th component of the parameter omitted.
+
+`os.final` with its j'th component omitted is used as the initial parameter.
+"""
+function optsumj(os::OptSummary, j::Integer)
+    return OptSummary(
+        deleteat!(copy(os.final), j),
+        deleteat!(copy(os.lowerbd), j),
+        os.optimizer
+    )
+end
+
+function profileobj!(
+    m::LinearMixedModel{T}, θ::AbstractVector{T}, opt::Opt, osj::OptSummary
+) where {T}
+    isone(length(θ)) && return objective!(m, θ)
+    fmin, xmin, ret = NLopt.optimize(opt, copyto!(osj.final, osj.initial))
+    _check_nlopt_return(ret)
+    return fmin
+end
+
+function profileθj!(
+    val::NamedTuple, sym::Symbol, tc::TableColumns{T}; threshold=4
+) where {T}
+    (; m, fwd, rev) = val
+    optsum = m.optsum
+    (; final, fmin, lowerbd) = optsum
+    j = parsej(sym)
+    θ = copy(final)
+    lbj = lowerbd[j]
+    osj = optsum
+    opt = Opt(osj)
+    if length(θ) > 1      # set up the conditional optimization problem
+        notj = deleteat!(collect(axes(final, 1)), j)
+        osj = optsumj(optsum, j)
+        opt = Opt(osj)               # create an NLopt optimizer object for the reduced problem
+        function obj(x, g)
+            isempty(g) ||
+                throw(ArgumentError("gradients are not evaluated by this objective"))
+            for i in eachindex(notj, x)
+                @inbounds θ[notj[i]] = x[i]
+            end
+            return objective!(m, θ)
+        end
+        NLopt.min_objective!(opt, obj)
+    end
+    pnm = (; p=sym)
+    ζold = zero(T)
+    tbl = [merge(pnm, mkrow!(tc, m, ζold))]    # start with the row for ζ = 0
+    δj = inv(T(64))
+    θj = final[j]
+    θ[j] = θj - δj
+    while (abs(ζold) < threshold) && θ[j] ≥ lbj && length(tbl) < 100  # decreasing values of θ[j]
+        ζ = sign(θ[j] - θj) * sqrt(profileobj!(m, θ, opt, osj) - fmin)
+        push!(tbl, merge(pnm, mkrow!(tc, m, ζ)))
+        θ[j] == lbj && break
+        δj /= (4 * abs(ζ - ζold))   # take smaller steps when evaluating negative zeta
```

Comment:
why smaller steps for negative zeta?

Reply:
Because problems with lack of monotonicity in the reverse spline always occur with decreasing values of θ.  The value of  ζ flattens out as constrained components of θ approach zero.  Attempts to create the reverse spline often end up with wild swings up and down when trying to invert a pattern that has flattened.  This is a crude attempt to fix the situation by adding more points on the left hand side of the estimate.

Reply:
Oh right, you had mentioned that at some point.